grid2op/tests/test_ma.py

```diff
@@ -0,0 +1,518 @@
+
+
+
+class MATesterGlobalObs(unittest.TestCase):
+    def setUp(self) -> None:
+        
+        with warnings.catch_warnings():
+            warnings.filterwarnings("ignore")
+            self.env = make("l2rpn_case14_sandbox", test=True, _add_to_name="test_ma")
+
+        self.action_domains = {
+            'agent_0' : [0,1,2,3, 4],
+            'agent_1' : [5,6,7,8,9,10,11,12,13]
+        }
+        return super().setUp()
+    
+    def tearDown(self) -> None:
+        self.env.close()
+        return super().tearDown()
+    
+    def test_verify_domains(self):
+        """test the MultiAgentEnv._verify_domains method """
+        # it should test that :
+        # 1) the function works (does not throw an error) when the input domains are correct
+        # 2) the function throws an error when the input domains are wrong
+        # (the more "wrong" cases tested the better)
+        
+        action_domains = {
+            'agent_0' : 0,
+            'agent_1' : [5,6,7,8,9,10,11,12,13]
+        }
+        # above action domain should raise an error: "0" is not an iterable !
+        with self.assertRaises(DomainException) as de:
+            MultiAgentEnv(self.env, action_domains, _add_to_name="test_verify_domains_0")
+        
+        action_domains = {
+            'agent_0' : [0],
+            'agent_1' : [5,6,7,8,9,10,11,12,13]
+        }
+        # above action domain should raise an error: substations are not fully allocated !
+        with self.assertRaises(DomainException) as de:
+            MultiAgentEnv(self.env, action_domains, _add_to_name="test_verify_domains_1")
+            
+        action_domains = {
+            'agent_0' : [],
+            'agent_1' : list(range(self.env.n_sub))
+        }
+        # above action domain should raise an error: agents must have at least one substation !
+        with self.assertRaises(DomainException) as de:
+            MultiAgentEnv(self.env, action_domains, _add_to_name="test_verify_domains_2")
+            
+        action_domains = {
+            'agent_0' : [0,1,6,3, 4],
+            'agent_1' : [5,2,7,8,9,10,11,12,13]
+        }
+        # this domain is valid even if it is not connected
+        try:
+            MultiAgentEnv(self.env, action_domains, _add_to_name="test_verify_domains_3")
+        except DomainException:
+            self.fail("action_domains raised Domain Exception unexpectedly!")
+            
+        action_domains = {
+            'agent_0' : [0,1,6,3, 4, 5],
+            'agent_1' : [5,2,7,8,9,10,11,12,13]
+        }
+        # this domain is valid even if it is not connected
+        with self.assertRaises(DomainException) as de:
+            MultiAgentEnv(self.env, action_domains, _add_to_name="test_verify_domains_4")
+        
+    
+    def test_build_subgrids_action_domains(self):
+        """Tests that the action_domains are correctly defined 
+            in MultiAgentEnv._build_subgrid_from_domain method
+        """
```

Comment:
I'm not sure, it tests that the substations ids are correctly assigned, but it is far from testing the whole action domains

Reply:
Changed the comment !